libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2515,21 +2515,28 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		let <end> := <encodeArgs>(add(<pos>, 4) <argumentString>)
 
 		let <success> := <call>(<gas>, <address>, <?hasValue> <value>, </hasValue> <pos>, sub(<end>, <pos>), <pos>, <reservedReturnSize>)
+		let <returnDataSize> := <estimatedReturnSize>
+		<?supportsReturnData>
+			if lt(<returnDataSize>, returndatasize()) {
```

Comment:
I think this should be
```suggestion
			if gt(<returnDataSize>, returndatasize()) {
```
and it should be executed only if we do not have dynamic returndata.

Also I think this is better moved a bit further down. You may even use
```
<?dynamicReturnSize>
// what we alread yhave
<!dynamicReturnSize>
// your new code
</dynamicReturnSize>
```